src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -72,17 +72,32 @@ GcsServer::GcsServer(const ray::gcs::GcsServerConfig &config,
       is_stopped_(false) {
   // Init GCS table storage.
   RAY_LOG(INFO) << "GCS storage type is " << storage_type_;
+
   switch (storage_type_) {
+  case StorageType::UNKNOWN:
+    RAY_LOG(FATAL) << "Unexpected storage type: " << storage_type_;
+    break;
   case StorageType::IN_MEMORY:
-    gcs_table_storage_ = std::make_shared<InMemoryGcsTableStorage>(main_service_);
+    store_client_ = std::make_unique<ObservableStoreClient>(
+        std::make_unique<InMemoryStoreClient>(main_service_));
     break;
   case StorageType::REDIS_PERSIST:
-    gcs_table_storage_ = std::make_shared<gcs::RedisGcsTableStorage>(GetOrConnectRedis());
+    auto redis_client = std::make_unique<RedisClient>(GetRedisClientOptions());
+    auto status = redis_client->Connect(main_service_);
+    RAY_CHECK(status.ok()) << "Failed to init redis gcs client as " << status;
+
+    // Init redis failure detector.
+    gcs_redis_failure_detector_ = std::make_unique<GcsRedisFailureDetector>(
```

Comment:
Should we move this into the RedisClient? (Ok not doing it here).

Reply:
Did it in the latest commit